pkg/backends/websockets.go

```diff
@@ -29,13 +29,9 @@ type WebsocketDialer struct {
 
 // NewWebsocketDialer instantiates a new WebsocketDialer backend
 func NewWebsocketDialer(address string, tlscfg *tls.Config, extraHeader string, redial bool) (*WebsocketDialer, error) {
-	addrURL, err := url.Parse(address)
-	if err != nil {
-		return nil, err
-	}
 	wd := WebsocketDialer{
 		address:     address,
-		origin:      fmt.Sprintf("http://%s", addrURL.Host),
+		origin:      fmt.Sprintf("http://%s", address),
```

Comment:
I think the `address` parameter supplied to this function is expected to be a URL.  So you probably still have to parse out the host part to make an origin.  Maybe the scheme also needs to be http or https based on whether `address`'s URL scheme is `ws` or `wss`.